configmap/parse.go

```diff
@@ -144,6 +146,45 @@ func AsQuantity(key string, target **resource.Quantity) ParseFunc {
 	}
 }
 
+func AsNamespacedName(key string, target *types.NamespacedName) ParseFunc {
+	return func(data map[string]string) error {
+		var (
+			name      string
+			namespace string
+			raw       string
+			ok        bool
+		)
+
+		if raw, ok = data[key]; !ok {
+			return nil
+		}
+
+		v := strings.Split(raw, string(types.Separator))
+		switch len(v) {
+		// No '/' separator.
+		case 1:
+			name = v[0]
+		case 2:
+			name = v[1]
+			namespace = v[0]
```

Comment:
Why not set `target.Name[space]`

Reply:
I want to validate things prior to setting them on target - to avoid any side effects on validation failures

Reply:
i can move things around to make it a bit cleaner